connectors/src/connectors/webcrawler/temporal/activities.ts

```diff
@@ -0,0 +1,156 @@
+import { ModelId } from "@dust-tt/types";
+import { CheerioCrawler, Configuration } from "crawlee";
+import turndown from "turndown";
+
+import { dataSourceConfigFromConnector } from "@connectors/lib/api/data_source_config";
+import { upsertToDatasource } from "@connectors/lib/data_sources";
+import { Connector } from "@connectors/lib/models";
+import {
+  WebCrawlerConfiguration,
+  WebCrawlerFolder,
+} from "@connectors/lib/models/webcrawler";
+import {
+  reportInitialSyncProgress,
+  syncSucceeded,
+} from "@connectors/lib/sync_status";
+
+export async function crawlWebsiteByConnectorId(connectorId: ModelId) {
+  const connector = await Connector.findByPk(connectorId);
+  if (!connector) {
+    throw new Error(`Connector ${connectorId} not found.`);
+  }
+  const webCrawlerConfig = await WebCrawlerConfiguration.findOne({
+    where: {
+      connectorId,
+    },
+  });
+  if (!webCrawlerConfig) {
+    throw new Error(`Webcrawler configuration not found for connector.`);
+  }
+
+  const dataSourceConfig = dataSourceConfigFromConnector(connector);
+  let pageCount = 0;
+  let errorCount = 0;
+  const createdFolders = new Set<string>();
+
+  const crawler = new CheerioCrawler(
```

Comment:
Does this work with JS rendered pages?

Reply:
No, but for now we are not supporting them. We can if needed later on.

Reply:
My guess is that user expectation will be that it works. It's quite a bad experience to see nothing if your browser in incognito mode shows something no?

Reply:
Here is the rationale:
Our initial target are websites generated by some sort of content management system (FAQ, help center etc), from my tests, these are either server side rendered and optimized for SEO, or impossible to crawl because they have now way to detect buttons other than looking at the UI as a human.

So for now I think this will cover a lot of our cases, and we can evolve to something more complex later.
Loading JS website means waiting for the fetch() calls to complete, waiting on some CSS selectors, which we don't know about since the URL is user provided.